luigi_pipeline/seqr_loading_optimized.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import os
+import re
+
+import luigi
+import hail as hl
+
+from hail_scripts.v02.utils.elasticsearch_client import ElasticsearchClient
+from lib.hail_tasks import HailMatrixTableTask, HailElasticSearchTask, GCSorLocalTarget, MatrixTableSampleSetError
+from lib.model.seqr_mt_schema import SeqrSchema, SeqrVariantSchema, SeqrGenotypesSchema
+import seqr_loading
+# from gnomad_hail.utils import vep_or_lookup_vep
+
+logger = logging.getLogger(__name__)
+
+
+class SeqrValidationError(Exception):
+    pass
+
+class SeqrVCFToVariantHTTask(seqr_loading.SeqrVCFToMTTask):
+    """
+    Loads all annotations for the variants of a VCF into a Hail Table (parent class of MT is a misnomer).
+    """
+
+    def run(self):
+        mt = self.import_vcf()
+        mt = hl.split_multi_hts(mt)
+        if self.validate:
+            self.validate_mt(mt, self.genome_version, self.sample_type)
+        mt = HailMatrixTableTask.run_vep(mt, self.genome_version, self.vep_runner)
+        # We're now adding ref data.
+        ref_data = hl.read_table(self.reference_ht_path)
+        clinvar = hl.read_table(self.clinvar_ht_path)
+        hgmd = hl.read_table(self.hgmd_ht_path)
+
+        ht = SeqrVariantSchema(mt, ref_data=ref_data, clinvar_data=clinvar, hgmd_data=hgmd).annotate_all(
+            overwrite=True).select_annotated_mt().rows()
+
+        ht.describe()
+
+        ht.write(self.output().path, stage_locally=True)
+        hl.copy_log(self.output().path + '-logs.log')
```

Comment:
Is this always wanted? Or should it be behind some kind of debug/verbose flag?

Reply:
So this is something that I haven't come to a conclusion on. It's actually nice to have these logs because in the past when we've had bad run times, the hail team has asked for logs and we have to re-run. What I'm not sure about is the convention, it seems like if we tie it to the bucket that the job creates, that'd be the ideal location. But I don't think it's readily accessible from the code.

Reply:
👍maybe worth creating a bucket dedicated to logs so that data buckets don't get cluttered.